packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -225,6 +225,7 @@ protected override ConstructorProvider[] BuildConstructors()
             // handle sub-client constructors
             if (ClientOptionsParameter is null)
             {
+                _clientParameters = _subClientInternalConstructorParams;
```

Comment:
This is for sub-client, which stores all client parameter in `_subClientInternalConstructorParams` (which already calculated when create ClientProvider instance).

Reply:
hello @jorgerangel-msft  when it is sub-client (ClientOptionsParameter is null), we will use `_subClientInternalConstrutorParams` to `ClientParamters`. When it is normal client, we will use the required parameters in public constructor.
And `_ClientParameters` is lazy load, it will be set only when it is used.

so we will not move this assignment to the clientProvider constructor